BAL 3733 - Enrich violations column in web presence table

shanegrouber opened the pull request:
<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

- **New Features**
  - Renamed the display labels from "Violations" to "Findings" for enhanced clarity.
  - Expanded the filtering to include all findings, sorted by risk level and name.
  - Updated the table to show up to four findings with an indicator for additional items.

- **Refactor**
  - Streamlined the rendering logic and icon styling for a more intuitive interface.
  - Updated the data access key from `contentViolations` to `allViolations` for consistency.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

shanegrouber committed:
feat(MerchantMonitoring): enhance violation handling in monitoring table

- Add website structure violations with associated risk levels
- Change field name from contentViolations to allViolations
- Update violation display logic to utilize new structure and sort accordingly

shanegrouber committed:
refactor(MerchantMonitoringTable): remove redundant violation structure

- Remove the WEBSITE_STRUCTURE_VIOLATIONS constant
- Update violations handling in the render function
- Limit displayed violations to the first four with a counter for additional findings

shanegrouber committed:
fix(MerchantMonitoringTable): improve filtering of violations

- Update filter logic to exclude violations without a name or risk level
- Ensure reliable sorting of violation entries based on risk level and name

tomer-shvadron commented on apps/backoffice-v2/src/pages/MerchantMonitoring/components/MerchantMonitoringTable/columns.tsx:
```diff
@@ -156,11 +156,19 @@ export const useColumns = ({ isDemoAccount = false }) => {
+              el => el.id !== NO_VIOLATION_DETECTED_RISK_INDICATOR_ID && el.name && el.riskLevel,
+            )
+            .sort((a, b) => {
+              if (a.riskLevel === b.riskLevel) {
+                return (a.name ?? '').localeCompare(b.name ?? '');
+              }
+
+              return (a.riskLevel ?? '').localeCompare(b.riskLevel ?? '');
```
```suggestion
              return a.riskLevel === b.riskLevel 
                ? (a.name ?? '').localeCompare(b.name ?? '') 
                : (a.riskLevel ?? '').localeCompare(b.riskLevel ?? '');
```

shanegrouber committed:
refactor(MerchantMonitoringTable): simplify sorting logic in useColumns

- Streamline comparison logic for sorting by riskLevel and name
- Improve readability and maintainability of the sorting function

alonp99 committed:
Merge branch 'dev' into BAL-3733
